contracts/poap/src/contract.rs

```diff
@@ -0,0 +1,451 @@
+    }
+}
+
+fn execute_set_mint_enabled(
+    deps: DepsMut,
+    info: MessageInfo,
+    mint_enabled: bool,
+) -> Result<Response, ContractError> {
+    let mut config = CONFIG.load(deps.storage)?;
+
+    // Check that the sender is the admin
+    if info.sender != config.admin {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    config.mint_enabled = mint_enabled;
+    // Save the new configurations
+    CONFIG.save(deps.storage, &config)?;
+
+    let action = if mint_enabled {
+        ACTION_ENABLE_MINT
+    } else {
+        ACTION_DISABLE_MINT
+    };
+
+    Ok(Response::new()
+        .add_attribute("action", action)
+        .add_attribute("sender", info.sender))
+}
+
+fn execute_mint(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    action: &str,
+    recipient_addr: Addr,
+    bypass_mint_enable: bool,
+    check_authorized_to_mint: bool,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    let event_info = EVENT_INFO.load(deps.storage)?;
+
+    // Check if the event is started
+    if !event_info.is_started(&env.block.time) {
+        return Err(ContractError::EventNotStarted {
+            current_time: env.block.time,
+            start_time: event_info.start_time,
+        });
+    }
+
+    // Check if the event is ended
+    if event_info.is_ended(&env.block.time) {
+        return Err(ContractError::EventTerminated {
+            current_time: env.block.time,
+            end_time: event_info.end_time,
+        });
+    }
+
+    // Check if the mint is enabled
+    if !bypass_mint_enable && !config.mint_enabled {
+        return Err(ContractError::MintDisabled {});
+    }
+
+    // Check if who is performing the action is the minter
+    if check_authorized_to_mint && info.sender != config.minter && info.sender != config.admin {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Check per address limit
+    let recipient_mint_count = (MINTER_ADDRESS
+        .key(recipient_addr.clone())
+        .may_load(deps.storage)?)
+    .unwrap_or(0);
+
+    if recipient_mint_count >= config.per_address_limit {
+        return Err(ContractError::MaxPerAddressLimitExceeded {
+            recipient_addr: recipient_addr.to_string(),
+        });
+    }
+
+    // Get the nex poap id
+    let poap_id = NEXT_POAP_ID.may_load(deps.storage)?.unwrap_or(1);
+
+    // Create the cw721 message to send to mint the poap
+    let mint_msg = Cw721ExecuteMsg::Mint(MintMsg::<String> {
+        token_id: poap_id.to_string(),
+        owner: recipient_addr.to_string(),
+        token_uri: Some(event_info.base_poap_uri),
+        extension: event_info.event_uri,
+    });
+
+    let cw721_address = CW721_ADDRESS.load(deps.storage)?;
+    let response_msg = CosmosMsg::Wasm(WasmMsg::Execute {
+        contract_addr: cw721_address.to_string(),
+        msg: to_binary(&mint_msg)?,
+        funds: vec![],
+    });
```

Comment:
It can be simplified to:
```rust
wasm_execute(
            cw721_address,
            &mint_msg,
            info.funds,
        )
```

Reference: [wasm_execute](https://github.com/CosmWasm/cosmwasm/blob/9bf920b00b76f31f3c73ed3632cd07cbd6a94b7b/packages/std/src/results/cosmos_msg.rs#L218)